Add support for custom location

ruiiiijiiiiang opened the pull request:
### Is your pull request linked to an existing issue?
https://github.com/iinsertNameHere/catnap/issues/143

### What is your pull request about?:
Allows user specified location for weather fetching.

iinsertNameHere commented:
Hey, thx for your PR.. Unfortunately if you look at the tests, they failed to fetch the wether stat. I don't know why they did not  fail, will open an issue an look into it..

ruiiiijiiiiang commented:
Hi @iinsertNameHere, I figured out why the tests were failing. The test config location was "San Francisco", and the space was breaking the wttr.in url. I added a `replace(" ", "+")` as instructed by wttr.in and it is working now. As for the tests, I noticed that they were passing for me locally because the weather info was cached, so I added a `-n` to each test script to force a refresh. I'm still not sure why it was passing in the github workflow though, so you might need to further look into that.

iinsertNameHere commented:
@ruiiiijiiiiang Thanks, for looking in to it!
